source/blood/src/view.cpp

```diff
@@ -1158,6 +1158,66 @@ void viewDrawStats(PLAYER *pPlayer, int x, int y)
     viewDrawText(3, buffer, x, y, 20, 0, 0, true, 256);
 }
 
+#define kSBarNumberHealth 9220
+#define kSBarNumberAmmo 9230
+#define kSBarNumberInv 9240
+#define kSBarNumberArmor1 9250
+#define kSBarNumberArmor2 9260
+#define kSBarNumberArmor3 9270
+
+struct POWERUPDISPLAY
+{
+    int nTile;
+    float nScaleRatio;
+    int yOffset;
+    int remainingDuration;
+};
+
+void sortPowerUps(POWERUPDISPLAY* powerups) {
+    for (int i = 1; i < 5; i++)
+    {
+        for (int j = 0; j < 5-i; j++)
+        {
+            if (powerups[j].remainingDuration > powerups[j+1].remainingDuration)
+            {
+                POWERUPDISPLAY temp = powerups[j];
+                powerups[j] = powerups[j+1];
+                powerups[j+1] = temp;
+            }
+        }
+    }
+}
+
+void viewDrawPowerUps(PLAYER* pPlayer)
+{
```

Comment:
Add a toggle for this in menu

Reply:
Sure, it takes no time, I just wonder if it is needed at all? Why would anybody turn this off? It is like the audio restarting feature which is off now by default and that doesn't have a toggle either.

Reply:
Another thing what I was thinking about is the default value of the toggle. Should this be on or off be default? Because if it is off by default and somebody doesn't know about this setting then it would be a handicap for somebody in multiplayer.
The remaining time and the sprite for a power up is only visible when the player has picked up such power-up, and it doesn't cover out much of the screen.
If you still want a toggle, sure, what should be the default value of it?

Reply:
Enable it by default. Music restarting thing is somewhat very subtle change imo, so cvar was enough there